### PR TITLE
COMMERCE-10891 OOTB Account Role of Supplier to Manage Catalogue

### PR DESCRIPTION
**Pull request by dsitu**
https://issues.liferay.com/browse/COMMERCE-10891

### CODE REVIEW

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
@dsitu needs to be rebased. Thanks

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/permission/CommerceCatalogPermissionImpl.java [hunk 1]
```diff
@@ -145,7 +147,10 @@ private boolean _contains(
 				if (commerceCatalog.getAccountEntryId() ==
 						accountEntry.getAccountEntryId()) {
 
-					return true;
+					return _roleLocalService.hasUserRole(
+						permissionChecker.getUserId(),
+						permissionChecker.getCompanyId(),
+						AccountRoleConstants.ROLE_NAME_SUPPLIER, true);
```
Comment (dsitu) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/permission/CommerceCatalogPermissionImpl.java:153]: @luca-pellizzon I added in supplier role check to align with @brianikim's permission logic.

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:stop

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Review by luca-pellizzon — APPROVED** [review]

**Comment by dsitu** [comment]
I think the failure is unrelated to my changes.

**Pull request closed without merging**